lib/mangoex/api/base.ex

```diff
@@ -16,13 +16,20 @@ defmodule Mangoex.API.Base do
     ]
   end
 
-  def request(:post, path, body, token) do
+  def request(:post, path, body, token), do:
+    request(:post, path, body, token, nil)
+  def request(:post, path, body, token, idempotency) do
     data = body |> Poison.encode!()
+    headers =
+      [
+        {"Authorization", "Bearer #{token}"},
+        {"Content-Type", "application/json"}
+      ]
+    unless idempotency == nil,
+         do: headers
+             |> Enum.concat([{"Idempotency-Key", "#{idempotency}"}])
 
-    HTTPoison.post! api_url(path), data, [
-      {"Authorization", "Bearer #{token}"},
-      {"Content-Type", "application/json"}
-    ]
+    HTTPoison.post! api_url(path), data, headers
```

Comment:
The value of `heeader` here is the one defined at line 23. Data is immutable in Elixir, so when calling `Enum.concat` line 30, you did not mutate the value of `headers`, you created a new value, which you then do not use.


```elixir
def request(:post, path, body, token, idempotency) do                                                                                            
  data = body |> Poison.encode!()
  headers =
    [
      {"Authorization", "Bearer #{token}"},                                                                                                          
      {"Content-Type", "application/json"}
  ]                                                                                                                                              
  |> add_idempotency_header(idempotency)
 
  HTTPoison.post! api_url(path), data, headers
end

defp add_idempotency_header(headers, nil), do: headers

defp add_idempotency_header(headers, idempotency_key) do
  [{"Idempotency-Key", "#{idempotency_key}"} | headers]
end
```

Reply:
Oh sorry, my commit was not send when you was reviewing. have a look how I did, it looks the same but authentication header have been extracted as a function.